app/hooks/challenges/useChallenges.tsx

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from "@tanstack/react-query";
+import { apis } from "@/api/initializeApi";
+import { Unauthorized } from "@/errors/Unauthorized";
+import { extractIdFromToken } from "@/utils/jwt";
+
+export const useChallenges = (userId?: number) => {
+    const queryKey = userId ? ["challenges", {userId}] : ["challenges"];
+
```

Comment:
ok . dont use useChallenges = (userId?: number) <- this userId?

instead make it useChallenges = ()

the same about profile , with this you can reduce amount of code duplication 

instead allways pass userId into hook